File: app/src/main/java/com/returnnotfound/stain/di/module/AppModule.java
package com.returnnotfound.stain.di.module;

import android.content.Context;

import com.returnnotfound.stain.App;
import com.returnnotfound.stain.di.AppContext;

import javax.inject.Singleton;

import dagger.Binds;
import dagger.Module;

@Module
public abstract class AppModule {

  @Binds
  @AppContext
  @Singleton
  abstract Context bindAppContext(App app);
}

File: app/src/main/java/com/returnnotfound/stain/di/module/AppModule.kt
package com.returnnotfound.stain.di.module

import android.content.Context
import com.returnnotfound.stain.App
import com.returnnotfound.stain.di.AppContext
import dagger.Binds
import dagger.Module
import javax.inject.Singleton

@Module
abstract class AppModule {
  @Binds
  @AppContext
  @Singleton
  abstract fun bindAppContext(app: App): Context
}